src/Sema.zig

```diff
@@ -26419,6 +26432,9 @@ fn fieldCallBind(
             const decl = mod.declPtr(decl_idx);
             try mod.errNoteNonLazy(decl.srcLoc(mod), msg, "'{}' is not a member function", .{field_name.fmt(ip)});
         }
+        if (concrete_ty.zigTypeTag(mod) == .ErrorUnion) {
```

Comment:
The error message doesn't include a type so either this should be removed or the message should be changed. Seeing that this note does not appear to be tested anywhere I'd go with the former.

Reply:
https://github.com/ziglang/zig/pull/16461/commits/6f36e99899c6451bd3c1b8b8ef442247c48c529e

The test is test/cases/compile_errors/method_call_on_error_union.zig

Reply:
If you do what the error message I added says and do for example `(x catch unreachable).a()` the code compiles. That was the purpose of the commit.

Reply:
My bad, seems like I mistook the above error note as the actual error message.